File: AHome/11.01/XMLInterpreter/XMLInterpreter/ControlSelector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace XMLInterpreter
{
    internal static class ControlSelector
    {
        public static Control ControlCreation(string type)
        {
            Control control = null;

            switch (type)
            {
                case "Panel":
                    control = new Panel();
                    break;
                case "Menu":
                    control = new Panel();
                    break;
                case "ToolBar":
                    control = new Panel();
                    break;
                case "ToolBox":
                    control = new Panel();
                    break;
                case "StatusBar":
                    control = new Panel();
                    break;
                case "TabControl":
                    control = new TabControl();
                    break;
                case "TabPage":
                    control = new TabPage();
                    break;
                case "TextBox":
                    control = new TextBox();
                    break;
                case "Button":
                    control = new Button();
                    break;
                case "Label":
                    control = new Label();
                    break;
                default:
                    throw new ArgumentException();
            }

            return control;
        }

    }
}

File: AHome/13.01/Java/DomBuilder/src/XML_Parser/ControlSelector.java
package XML_Parser;

import java.awt.Color;
import java.awt.Component;

import javax.swing.JButton;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.JTabbedPane;
import javax.swing.JTextField;
import javax.swing.border.LineBorder;

public class ControlSelector {
	
	public static Component ControlCreation(String type)
    {
		Component control = null;

        switch (type)
        {
            case "Panel":
                control = new JPanel();
                ((JPanel)control).setLayout(null);
                break;
            case "Menu":
                control = new JPanel();
                ((JPanel)control).setLayout(null);
                break;
            case "ToolBar":
                control = new JPanel();
                ((JPanel)control).setLayout(null);
                break;
            case "ToolBox":
                control = new JPanel();
                ((JPanel)control).setLayout(null);
                break;
            case "StatusBar":
                control = new JPanel();
                ((JPanel)control).setLayout(null);
                break;
            case "TabControl":
                control = new JTabbedPane();
                break;
            case "TextBox":
                control = new JTextField();
                break;
            case "Button":
                control = new JButton();
                break;
            case "Label":
                control = new JLabel();
                break;
            case "TabPage":
            	control = new JPanel();
            	control.setBackground(Color.YELLOW);
            	break;
            default:
            	control = null;
            	break;
        }

        return control;
    }
	
}
